pkg/crd/markers/validation.go

```diff
@@ -159,6 +161,14 @@ type Type string
 // This is often not necessary, but may be helpful with custom serialization.
 type Nullable struct{}
 
+// +controllertools:marker:generateHelp:category="CRD validation"
+// Default sets the default value for this field.
+//
+// A default value will be accepted as a json-formatted string (e.g. `true`,
+// `"Cluster"`, `1.24`, `[1,2]`, `{"policy": "delete"}`) and validated as the
+// type indicated for the field.
+type Default string
```

Comment:
as we discussed, would `type Default struct { interface{} }` work?

Reply:
That is not a valid golang type definition. It would be necessary to define the empty interface as a named field rather than embedding it, requiring either that the field name be provided (e.g. `Default=Value=<value>`) or that the parser be special-cased to not require the field name. `AnyType` parsing currently lacks support for object parsing (e.g. `{"foo": "bar"}`), and I'm not sure it's worth implementing for defaulting since the raw argument can be trivially json unmarshalled.

Instead, I defined `Default` as `markers.RawArguments` (`[]byte`) and revised marker construction to enable the use of `RawType` parsing. The previous strategy to detect when a field should be parsed as `RawType` relied upon comparing the field type to `RawArguments`. This works for a nested type (e.g. slice element or struct field) since the type can be identical, but a user-defined type fails that check (e.g. `reflect.TypeOf(Default) != reflect.TypeOf(RawArguments)`).